daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/processors/unparsers/SeparatedSequenceUnparsers.scala

```diff
@@ -260,7 +260,6 @@ class OrderedSeparatedSequenceUnparser(
           val erd = unparser.erd
           var numOccurrences = 0
           val maxReps = unparser.maxRepeats(state)
-          val isBounded = unparser.isBoundedMax
```

Comment:
Notice this variable was never used. Unrelate to the fix, but thought I'd clean this up and remove it while I'm here.

Reply:
Could be a bug that it is unused.

Reply:
It loosk like ``isBounded`` is only used in ``unparsePositionallyRequiredSeps``, which is only called from ``unparseWithSuppression``. I'm thinking that because ``separatorSuppressionPolicy="never"`` that there is no need for positionally required separators, and no need to check if the occurrences are bounded? Note that ``unparseWithNoSuppression`` does look at the ``maxReps`` to determine if extra separators are needed.

Reply:
This is an interesting thing. 

When refactoring this code I didn't create a constructor that required this as an argument. It is something needed by the general case, not by special cases. In some sense the special cases should still have all the parameters that the general case requires, or maybe should have them as Optional[Thing] and then just not use some of them. This falls in that category.  

To me we lose something by deleting this line. The fact that isBounded isn't needed or used by this case, should somehow be explicitly stated. The classes as formulated don't force us to do this by providing None for an Option[Boolean]. 

I guess my suggestion is to leave this line in, but comment out this line and add a line-end comment that says unused in this case.

Reply:
Looking at the again, the ``val isBounded`` that is unused and removed above was actualy from the ``unparseWithSuppression`` case, which does end up using it when it calls ``unparsePositionallyRequiredSeps``, it's just not passed in.

So probably what happened is that ``isBounded`` was used at some point in this function, but that logic got refactored out to a function, and the variable just wasn't removed.

I do agree that it's probably still valuable to point out that isBounded is not used in the no-suppression case, so I'll add that and comment it out. But I think this line above can stay removed, since isBounded does apply, just via another function.